controllers/managedfusion_controller.go

```diff
@@ -709,6 +709,7 @@ func (r *ManagedFusionReconciler) reconcileK8SMetricsServiceMonitor() error {
 		}
 		desired := templates.K8sMetricsServiceMonitorTemplate.DeepCopy()
 		r.k8sMetricsServiceMonitor.Spec = desired.Spec
+		utils.AddLabel(r.k8sMetricsServiceMonitor, monLabelKey, monLabelValue)
```

Comment:
- why was this not required in deployer but needed in agent?

Reply:
We are doing the same in deployer but differently. 
Checkout reconcileMonitoringResources function, we do add labels to podMonitors, serviceMonitors and prometheusRules. In Agent, we moved that piece of code in dataFoundation reconciler and didn't think that this is also required for the k8s-metric-service-monitor.
Ref: https://github.com/red-hat-storage/ocs-osd-deployer/blob/main/controllers/managedocs_controller.go#L1395-L1435

Reply:
I meant the label (app: ...-agent) for service monitor as I was under the impression service monitor doesn't require to be labelled.

let me know if I'm wrong.

Reply:
The prom instance only selects resources that have a label matching the label selector. 
This is how we configure our prom the select monitoring resources.

Reply:
@leelavg we have labelSelector for serviceMonitor, podMonitor and promRules in prometheus CR. So our prometheus instance selects the resources based on label.